aclk/aclk_otp.c

```diff
@@ -14,15 +14,19 @@ static int aclk_https_request(https_req_t *request, https_req_response_t *respon
     // wrapper for ACLK only which loads ACLK specific proxy settings
     // then only calls https_request
     struct mqtt_wss_proxy proxy_conf = { .host = NULL, .port = 0, .username = NULL, .password = NULL, .type = MQTT_WSS_DIRECT };
-    aclk_set_proxy((char**)&proxy_conf.host, &proxy_conf.port, &proxy_conf.type);
+    aclk_set_proxy((char**)&proxy_conf.host, &proxy_conf.port, (char**)&proxy_conf.username, (char**)&proxy_conf.password, &proxy_conf.type);
 
     if (proxy_conf.type == MQTT_WSS_PROXY_HTTP) {
         request->proxy_host = (char*)proxy_conf.host; // TODO make it const as well
         request->proxy_port = proxy_conf.port;
+        request->proxy_username = proxy_conf.username;
+        request->proxy_password = proxy_conf.password;
```

Comment:
Hello @underhood ,

While I was compiling I observed these warnings:

```sh
aclk/aclk_otp.c: In function 'aclk_https_request':
aclk/aclk_otp.c:22:33: warning: assignment discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
   22 |         request->proxy_username = proxy_conf.username;
      |                                 ^
aclk/aclk_otp.c:23:33: warning: assignment discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
   23 |         request->proxy_password = proxy_conf.password;
      |                                 ^
 OK  ''
```

Best regards!

Reply:
The PR is working as expected, please, take a look in this warning.

Reply:
thank you it should be resolved, currently CI seems to have some unrelated issues